### PR TITLE
Revert #540, add migration to revert related data migration

### PR DESCRIPTION
**Pull request by noisecapella**
This:
- adds a migration to delete curation vocabularies and terms assigned to learning resources
- removes code related to curation vocabularies
- adjusts tests accordingly

Changes to `get_preview_url` were kept as well as unrelated test infrastructure changes

### CODE REVIEW

File: roles/signals.py [hunk 1]
```diff
@@ -45,6 +45,3 @@ def add_creator_permission(sender, **kwargs):
         instance,
         GroupTypes.REPO_ADMINISTRATOR,
     )
-    # This signal is only fired upon initial creation of a repository,
-    # so receivers don't have to check whether the repository is new.
-    repo_created.send(sender=instance)
```
Comment (pwilkins) [inline roles/signals.py:50]: Why is the signal removed?  Was it created only for the curation status vocab?  It sounds like a useful signal though I suppose we don't need to add signals that we don't use.

Comment (noisecapella) [inline roles/signals.py:50]: It's only used in creating default vocabularies so without that functionality there is no need for it. We can add it back in the PR implementing this functionality again

Comment (pwilkins) [inline roles/signals.py:50]: Ah, I just read the taxonomy/signals/**init**.py file docstring.

File: learningresources/migrations/0016_revert_backfill.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.utils.text import slugify
+
+# pylint: skip-file
+
+VOCAB_NAME = "curation status"
+EMPTY_VALUE = "--not set--"
```
Comment (bdero) [inline learningresources/migrations/0016_revert_backfill.py]: It looks like EMPTY_VALUE isn't used, can you remove it?

**Comment by giocalitri** [comment]
Functionality from `master` to `PR commit` is fine: terms and extra vocabulary has been removed

Testing functionality from `0.9` to `PR commit`

File: taxonomy/tests/test_models.py [hunk 3]
```diff
@@ -54,24 +54,24 @@ def test_slug(self):
...
     """
-    Tests for the automatically-created curator vocabulary.
+    Tests for automatically-created vocabularies.
     """
     def test_vocab_created(self):
         """
-        A Repository should have the "curation status" vocabulary added
-        automatically upon creation.
+        At the moment there are no vocabularies automatically created
+        though that will change soon.
```
Comment (pwilkins) [inline taxonomy/tests/test_models.py:62]: Consider mentioning issue #576 as the description of the change.  (Though I suppose this docstring will be short-lived.)

**Comment by bdero** [comment]
The migration looks straightforward and clean - :+1: after @giocalitri 's review

File: taxonomy/tests/test_models.py [hunk 4]
```diff
@@ -54,24 +54,24 @@ def test_slug(self):
...
         )
-        self.assertEqual(len(get_vocabularies(repo.id)), 1)
+        self.assertEqual(len(get_vocabularies(repo.id)), 0)
 
     def test_new_resource(self):
         """
-        When a new LearningResource is created, it should automatically
-        have the default (not set) value of the curator status vocabulary.
+        When a new LearningResource is created, it will currently have no
+        terms assigned to it.
```
Comment (pwilkins) [inline taxonomy/tests/test_models.py:74]: nit: Consider removing "currently" since it makes the docstring ambiguous.

**Comment by giocalitri** [comment]
functionality from `0.9` to `PR commit` is fine: going through the code now

**Comment by giocalitri** [comment]
code is fine: :+1:

**Comment by pwilkins** [comment]
LGTM, too.

**Pull request merged**